R/module_teal_with_splash.R

```diff
@@ -22,14 +22,17 @@ ui_teal_with_splash <- function(id,
                                 title,
                                 header = tags$p("Add Title Here"),
                                 footer = tags$p("Add Footer Here")) {
-  checkmate::assert_multi_class(data, c("TealDataAbstract", "teal_data"))
+  checkmate::assert_multi_class(data, c("TealData", "teal_data", "teal_data_module"))
   ns <- NS(id)
 
   # Startup splash screen for delayed loading
   # We use delayed loading in all cases, even when the data does not need to be fetched.
   # This has the benefit that when filtering the data takes a lot of time initially, the
   # Shiny app does not time out.
-  splash_ui <- if (inherits(data, "teal_data")) {
+
+  splash_ui <- if (inherits(data, "teal_data_module")) {
+    data$ui(ns("data"))
```

Comment:
something more specific?
```suggestion
    data$ui(ns("teal_data_module"))
```